### PR TITLE
Add time-related types to the Admin and Schema loader

### PR DESCRIPTION
**Pull request by Torch3333**
## Description

This updates the Admin and Schema Loader components to be able to manage schema with the new time-related types:
- DATE
- TIME
- TIMESTAMP
- TIMESTAMPTZ

> **_NOTE:_** Cassandra does not support the TIMESTAMP type but we plan to support it in the future.

At the moment, I plan to have a second PR which includes the CRUD interface changes, and a third one which includes the changes to the import table feature.

## Related issues and/or PRs

N/A

## Changes made

Update the implementations of the `DistributedStorageAdmin` and Schema Loader to support managing schema with time-related types.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

This is the first PR of several PR to support time-related on ScalarDB core, the changes of this feature will be merged on a feature branch [add_time_related_types](https://github.com/scalar-labs/scalardb/tree/add_time_related_types) before being merged into the master branch.

## Release notes

N/A

### CODE REVIEW

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminRepairIntegrationTest.java [hunk 1]
```diff
@@ -28,4 +28,8 @@ protected void initialize(String testName) {
     admin = new CassandraAdmin(clusterManager, new DatabaseConfig(properties));
     adminTestUtils = new CassandraAdminTestUtils(properties, clusterManager);
   }
+
+  protected boolean isTimestampTypeSupported() {
```
Comment (feeblefakie) [inline core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminRepairIntegrationTest.java:32]: `@Override`  is missing?

**Review by feeblefakie — APPROVED** [review]
Although I left one comment, LGTM! Thank you!

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminRepairIntegrationTest.java (hunk 1 above)
Comment (Torch3333) [reply core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminRepairIntegrationTest.java:33]: Indeed, thank you. 
I revised it in https://github.com/scalar-labs/scalardb/pull/2301/commits/2ec00e4f705e12c30b9b52f8f830e6b06ba45314

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

File: core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java [hunk 2]
```diff
@@ -203,6 +203,13 @@ public String getDataTypeForEngine(DataType scalarDbDataType) {
         return "INT";
       case TEXT:
         return "LONGTEXT";
+      case DATE:
+        return "DATE";
+      case TIME:
+        return "TIME";
+      case TIMESTAMP:
```
[the remark this answers is not in the document]
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java:210]: Yes, your assumption is correct, the time will be stored in UTC.
MySQL does not provide a type to store timestamp that is timezone or offset-free. We could choose to store ScalarDB TIMESTAMP in plaintext using a VARCHAR but using the DATETIME or TIMESTAMP seems more intuitive in my opinion. Also, MYSQL has built-in features for these types that maybe useful for some users such as auto-updating these columns (https://dev.mysql.com/doc/refman/8.4/en/timestamp-initialization.html)

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Pull request merged**